### DISCUSSION TITLE
Slack webhook and alerts

### DISCUSSION THREAD

**Discussion by michbsd** [fluxcd/flux2#2575] [Q&A] [answered]
Hi, 

I have set up a slack alert provider, and it is working -  get alerts as such: 

![Screenshot 2022-03-25 at 13 05 52](https://user-images.githubusercontent.com/8202334/160117766-b69d6902-e6b9-41c0-a7c4-f333791839bf.png)
 
The above alert is after I have pushed a new image release in the flux repo. 

However, I would like two things:
- when a new version of my helm chart is available
- version information about the image version in the new helmrelease

Is it possible?

**Comment by kingdonb** [comment]
This is not currently possible through Alerts; you can get an alert about a HelmRelease when it is upgrading to a new version, and that will include the chart version information, but HelmRepository does not keep track of that (image / app version and chart version). It only makes those versions accessible to the HelmRelease, should one of them be selected by the semver rule or `chart.spec.version`.

There are notifications for when a new index is loaded, the HelmRepository emits a notification when it reconciles, however this does not contain the version information for either the app or chart – HelmRepository does not know which chart versions are installable.

Could you say more about your use case here? Are you using the automation, or are you intending to use these notifications to help you decide whether to upgrade to the next version or not?

**Comment by michbsd** [comment, marked as answer]
Thanks for your reply, @kingdonb 
When I make a commit to my flux repo changing the image tag, e.g.

```
  values:
    image:
      tag: 1.10.0
```

Once this is reconciled, I get the following slack notification (twice, for some reason):

```
kustomization/flux-system.flux-system
HelmRelease/xxx/xxx configured
revision
master/4c8acd808f3a2a51aa6a1589xxxxxxxxxxxxxxxx
```

I would like my developers to get a notification that the deployment has been updated with the new image and its version. 

Hope this is clear enough. 


dddf

**Reply by kingdonb** [reply]
Thanks that does help clarify, and that does seem like a pretty straightforward ask from the perspective of ImageUpdateAutomation users who are also Helm Controller users. 👍

**Reply by stefanprodan** [reply]
Add the `ImageUpdateAutomation` object to the alert and the message will contain the image tag. Flux alerts for image automation contain the same text as the the commit message.
